CHANGELOG

```diff
@@ -1,3 +1,48 @@
+4.0.0.pre4 (153 commits)
+-------------------------------
+
```

Comment:
I'd add a short summary here to give a broad overview :) I think the biggest story this time was performance, mostly thanks to @jasonrclark ! :)